src/routes/teams.rs

```diff
@@ -18,8 +21,136 @@ pub async fn team_members_get(
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+    let current_user = get_user_from_headers(req.headers(), &**pool).await.map_err(|_| ApiError::AuthenticationError)?;
+
+    TeamMember::edit_team_member(team_id, current_user.id.into(), None, None, Some(true), &**pool).await?;
+
+    Ok(HttpResponse::Ok().body(""))
+}
+
+#[post("{id}/members")]
+pub async fn add_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+    new_member: web::Json<crate::models::teams::TeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+
+    let mut transaction = pool.begin().await.map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    let current_user = get_user_from_headers(req.headers(), &**pool).await.map_err(|_| ApiError::AuthenticationError)?;
+    let team_member = TeamMember::get_from_user_id(team_id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if (member.permissions & (1 << 4)) != 0 && new_member.role != crate::models::teams::OWNER_ROLE {
+            // TODO: Prevent user from giving another user permissions they do not have
+            let new_id = crate::database::models::ids::generate_team_member_id(&mut transaction).await?;
+            TeamMember {
+                id: new_id,
+                team_id,
+                user_id: new_member.user_id.into(),
+                name: new_member.name.clone(),
+                role: new_member.role.clone(),
+                permissions: new_member.permissions as i64,
+                accepted: false
+            }.insert(&mut transaction).await.map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            transaction.commit().await.map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            Ok(HttpResponse::Ok().body(""))
+        } else {
+            Err(ApiError::AuthenticationError)
+        }
+    } else {
+        Err(ApiError::AuthenticationError)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct EditTeamMember {
+    pub permissions: Option<u64>,
+    pub role: Option<String>,
+}
+
+#[patch("{id}/members/{user_id}")]
+pub async fn edit_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId, UserId)>,
+    pool: web::Data<PgPool>,
+    edit_member: web::Json<EditTeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let ids = info.into_inner();
+    let id = ids.0.into();
+    let user_id = ids.1.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool).await.map_err(|_| ApiError::AuthenticationError)?;
+    let team_member = TeamMember::get_from_user_id(id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if (member.permissions & (1 << 6)) != 0 && edit_member.role.clone().unwrap_or("".to_string()) != "Owner".to_string() {
+            // TODO: Prevent user from giving another user permissions they do not have
```

Comment:
The same as the above, this shouldn't be merged until this is implemented.  This should also check the role, since if "owner" is still special cased, the new role could be "owner", so multiple team members would be marked as the owner.
Should we have a special system for transferring ownership of a team?

Reply:
A lowercase owner is fine, for consistency I would like any owner of a mod for their role to be "Owner". We can add an ownership transfer soon

Reply:
If this is fixed, this TODO should be removed